parser: invaild memory address

```sql
WORKSPACE FiscalCloudWS (

    DESCRIPTOR FiscalCloud (
        Containers FiscalContainerItem
    );

    TABLE FiscalContainerItem INHERITS CRecord(
        ChildWorkspaceID ref,
        WLogOffs int64,
        Customers int32
    );
)
```

error:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
```

--- a/pkg/parser/impl_analyse.go
+++ b/pkg/parser/impl_analyse.go
@@ -836,11 +836,19 @@ func analyseWorkspace(v *WorkspaceStmt, c *iterateCtx) {
 		}
 	}
 	if v.Descriptor != nil {
+		wc := &iterateCtx{
+			basicContext: c.basicContext,
+			collection:   v,
+			pkg:          c.pkg,
+			parent:       c,
+			wsCtxs:       c.wsCtxs,
+		}
 		if v.Abstract {
 			c.stmtErr(&v.Descriptor.Pos, ErrAbstractWorkspaceDescriptor)
 		}
-		analyseNestedTables(v.Descriptor.Items, appdef.TypeKind_CDoc, c)
-		analyseFieldSets(v.Descriptor.Items, c)
+		analyseNestedTables(v.Descriptor.Items, appdef.TypeKind_CDoc, wc)
+		analyseFields(v.Descriptor.Items, wc, true)
+		analyseFieldSets(v.Descriptor.Items, wc)
 	}
 
 	// find all included QNames

--- a/pkg/parser/impl_test.go
+++ b/pkg/parser/impl_test.go
@@ -2475,3 +2475,19 @@ func Test_UniquesFromFieldsets(t *testing.T) {
 	require.NoError(err)
 	require.NoError(BuildAppDefs(schema, appdef.New()))
 }
+
+func Test_CRecordInDescriptor(t *testing.T) {
+	require := assertions(t)
+	schema, err := require.AppSchema(`APPLICATION test();
+	WORKSPACE w (
+		DESCRIPTOR wd(
+			items x
+		);
+		TABLE x INHERITS CRecord(
+			f1 int32
+		);
+	);
+`)
+	require.NoError(err)
+	require.NoError(BuildAppDefs(schema, appdef.New()))
+}

